src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/process/ProcessRunner.java

```diff
@@ -14,11 +14,15 @@
 
 package com.google.cloud.tools.app.impl.cloudsdk.internal.process;
 
+import java.util.Map;
+
 /**
  * Interface for running arbitrary commands.
  */
 public interface ProcessRunner {
 
   void run(String[] command) throws ProcessRunnerException;
 
+  void run(String[] command, Map<String, String> environment) throws ProcessRunnerException;
```

Comment:
I'm not sure about overloading the run method. Maybe we can just add a `setEnvironment(Map<String, String> environment)`?